tmk_core/common/chibios/eeprom_wb32.c

```diff
@@ -0,0 +1,260 @@
+/*
```

Comment:
Does this add any value outside implementing a flash abstraction and reusing the stm wear leveling code?

Reply:
Resolved

Reply:
> Resolved

Not really, as you still not given justification to why it cannot use the existing wear leveling code or swapped it over to use said code.

Reply:
Because the flash operation of the WB32 MCU is different from other MCU operations

Reply:
"different" isnt really answering why it cant reuse the existing code within `eeprom_stm` and implement its own `flash_wb` logic.

From `EEPROM_WriteDataByte`
> If a byte is already written, the whole page must be copied to a buffer, the byte changed and the manipulated buffer written after PageErase.

```c
    WBFLASH_Read(pagpos * FEE_PAGE_SIZE + WB32_FLASH_BASE, wb_flash_buffer, FEE_PAGE_SIZE);
    wb_flash_buffer[pagoff] = DataByte;
    /* Erase the specified FLASH page */
    FMC_ErasePage(pagpos * FEE_PAGE_SIZE + WB32_FLASH_BASE);
```

Thats not what this code actually does, however the implications are actually worse. The flash wear cylcles are unnecessarily burnt. Using existing code where possible ensures issues like this are potentially mitigated.

Reply:
As per the datasheet the flash has 100k write cycle endurance.
If writing a single effective byte to the flash page requires reading 256 bytes, then writing 256 bytes, the endurance may potentially be exceeded within a few weeks, potentially corrupting any stored data and/or preventing any future data storage.

Suggest re-thinking the decision to not integrate with the existing wear-leveling code, which erases pages only when required.

Reply:
In fact, in the reference manual, section 2.4:

* The flash should be program by page (256 byte), byte program operation is not supported.
* The flash page should be erased before program, it is not recommend to program the same page multiple times without erase.

Can you clarify the second point? Given these two statements as they stand, I'm inclined to reject the inclusion of this emulated EEPROM implementation entirely, given that the likelihood of failure is going to be very high in a short period of time.

Reply:
WB32 flash has 200k+ write cycle endurance.
I have modified the code, please check whether there are other problems.

Reply:
If it doesn't pass, can I delete this part of the code and pass?

Reply:
> Can you clarify the second point?

Can you actually answer rather than marking this as resolved.

Reply:
We can see that from the datasheet.

The issue is that if we go ahead with this current implementation, the effective lifetime of the pages used will be severely diminished, given that the number of write cycles for the same byte locations is limited to 200k (despite the datasheet saying 100k).

Every write to every location results in the read/erase/write of a whole page of 256 bytes. Given that the vast majority of core QMK EEPROM usage occurs in the first 35 bytes, the first page may be rendered unusable within a few weeks, depending on whether the user regularly modifies their RGB settings for instance.

If we allow this implementation to go into the repository, any keyboard manufacturer using them will need to be informed that they may see high failure rates.

Reply:
I'll remove the code related to the eeprom_wb32 first.

Reply:
> 
> 
> > Can you clarify the second point?
> 
> Can you actually answer rather than marking this as resolved.

This is determined by our Flash technical craft.

Reply:
@wb-Joy has corresponded with me and wanted me to state that they:
want the see this PR merged. To achieve that they want to not submit this part of the code for now, and submit this at a later time after this breaking change as an update.

>Flash technical craft
by this they mean flash fabrication technology